server/src/api/discounts/discounts.router.ts

```diff
@@ -35,6 +36,25 @@ discountsRouter.get('/', checkJwt, checkScopes, async(
 });
 
 
+/**
+ * route: GET /id
+ *
+ * @type {?}
+ */
+discountsRouter.get('/:code', async(
+    req: Request,
+    res: Response,
+) => {
+    try{
+      const codes = await checkDiscountCode(req.params.code);
```

Comment:
This is the kind of place were we should use query parameters rather then path parameters since the function of this route is query the data based on the passed value. You could set the endpoint to be a new path (something like `findcode` or whatever) and then use `req.query.code` to get any query parameters that were passed in. Here's the idea - [https://stackabuse.com/get-query-strings-and-parameters-in-express-js/](https://stackabuse.com/get-query-strings-and-parameters-in-express-js/)